src/SignalRServiceExtension/Config/ServiceManagerStore.cs

```diff
@@ -41,21 +45,26 @@ public IInternalServiceHubContextStore GetByConfigurationKey(string connectionSt
 
         private IInternalServiceHubContextStore CreateHubContextStore(string connectionStringKey)
         {
-            return new ServiceCollection().AddSignalRServiceManager()
+            var services = new ServiceCollection()
                 .WithAssembly(Assembly.GetExecutingAssembly())
                 .SetupOptions<ServiceManagerOptions, OptionsSetup>(new OptionsSetup(configuration, loggerFactory, connectionStringKey))
                 .PostConfigure<ServiceManagerOptions>(o =>
                 {
-                    if (string.IsNullOrWhiteSpace(o.ConnectionString))
+                    if ((o.ServiceEndpoints == null || o.ServiceEndpoints.Length == 0) && string.IsNullOrWhiteSpace(o.ConnectionString))
                     {
                         throw new InvalidOperationException(ErrorMessages.EmptyConnectionStringErrorMessageFormat);
                     }
                 })
+                .AddSignalRServiceManager()
                 .AddSingleton(loggerFactory)
-                .AddSingleton(configuration)
-                .AddSingleton<IInternalServiceHubContextStore, ServiceHubContextStore>()
-                .BuildServiceProvider()
-                .GetRequiredService<IInternalServiceHubContextStore>();
+                .AddSingleton<IInternalServiceHubContextStore, ServiceHubContextStore>();
+            if (router != null)
+            {
+                services.AddSingleton(router);
+            }
+            services.AddSingleton(services.ToList() as IReadOnlyCollection<ServiceDescriptor>);
```

Comment:
what's this for?

Reply:
This service is required for the building of service manager. Just updated in SDK.